### ISSUE TITLE
The meditation center in Open Sangha Map still points to the old profile page

### ISSUE THREAD

**Issue by coiby** [OpenSanghaFoundation/OSF#53]
**Describe the bug**

**To Reproduce**
1. Go to https://opensanghafoundation.org/newsite/monasticsangha-2/
2. Click on one of the places e.g. Santi Janati Costa Rica
3. Scroll down to the end to find the "Details" field  
4. The url points to https://opensanghafoundation.org/newsite/test-2/user-update/?r=2125 instead of https://opensanghafoundation.org/newsite/user/santi+janati/

**Expected behavior**

The Details field should point to the profile page.

**Screenshots**
![Screenshot_2024-07-10_20-31-32](https://github.com/OpenSanghaFoundation/OSF/assets/1511656/2907639d-4bc2-4e4e-835f-1d9077f72c11)

**Additional context**

https://opensanghafoundation.org/newsite/test-2/user-update currently is supposed to be only used by administrator and will be retried soon.

**Comment by coiby** [comment]
We can a simple fix or we can re-work the Sangha map page by using this Ultimate Member extension https://ultimatemember.com/extensions/user-locations/ which has some nice features like searching users by location. I suggest @anuutama to take a look at this issue if he doesn't have one in mind.

**Comment by OSFOSF** [comment]
the url points to the profile just click on the name, if yo click on the photo it will fo the the user updat
    On Wednesday, July 10, 2024 at 07:35:08 PM GMT+7, Coiby ***@***.***> wrote:  
 
 


Describe the bug
A clear and concise description of what the bug is.

To Reproduce
Steps to reproduce the behavior:
   
   - Go to https://opensanghafoundation.org/newsite/monasticsangha-2/
   - Click on one of the places e.g. Santi Janati Costa Rica
   - Scroll down to the end to find the "Details" field
   - The url points to https://opensanghafoundation.org/newsite/test-2/user-update/?r=2125 instead of https://opensanghafoundation.org/newsite/user/santi+janati/

Expected behavior

The Details field should point to the profile page.

Screenshots
Screenshot_2024-07-10_20-31-32.png (view on web)

Additional context

https://opensanghafoundation.org/newsite/test-2/user-update currently is supposed to be only used by administrator will be retried soon.

**Comment by anuutama** [comment]
As the https://opensanghafoundation.org/newsite/test-2/user-update site is only accessible to administrators, it redirects non-admin users to the normal user profile (https://opensanghafoundation.org/newsite/user/santi+janati/). I think non-admins user may have to wait a few more miliseconds for the redirections.

I notice that the map is created using GoogleMyMaps, so the user location data (osf_googlemap) are exported and the profile links are likely hard-coded into the GogoleMyMaps editors. Another question that I have is. since all users also have osf_googlemap metadata, how are the monasteries selected to each respective layers? Was it manually performed?

The UM user-locations extension https://ultimatemember.com/extensions/user-locations/ looks good, and requires google map APIs, while the current GoogleMyMaps is free(?). I am not sure how to integrate the layering information, i.e. Wat, Temples, Mahayana, Zen, etc, into UM's user location. Currently there is an option to display user roles marker, i.e. subscribers, admin, etc, and we may have to change them into something like "practitioners", "monk", "wat", "zen temples", etc.

Let me know if it would be good to develop UM's user location, I think I can help on that. For the current issue, I think I would need an access to the GoogleMyMaps, or its associated csv files. I can perform a mapping from "user-update/?r=2125" to "user/santi-janati" and all other locations. Else, we just let the non-admin members wait a bit longer :P 

Let me know what you think!

**Comment by coiby** [comment]
> the url points to the profile just click on the name, if yo click on the photo it will fo the the user updat

@OSFOSF  I can't find a name field that's clickable. And clicking  profile photo url simply bring me to the photo itself. Or do I miss anything?

**Comment by coiby** [comment]
> As the https://opensanghafoundation.org/newsite/test-2/user-update site is only accessible to administrators, it redirects non-admin users to the normal user profile (https://opensanghafoundation.org/newsite/user/santi+janati/). I think non-admins user may have to wait a few more miliseconds for the redirections.
> 
> I notice that the map is created using GoogleMyMaps, so the user location data (osf_googlemap) are exported and the profile links are likely hard-coded into the GogoleMyMaps editors. Another question that I have is. since all users also have osf_googlemap metadata, how are the monasteries selected to each respective layers? Was it manually performed?
> 
> The UM user-locations extension https://ultimatemember.com/extensions/user-locations/ looks good, and requires google map APIs, while the current GoogleMyMaps is free(?). 

Thanks for investigating this issue! Hi Dhammarato (@OSFOSF), if this Open Sangha Map page is created by you, can you help answer Adrian's questions? 

As  for cost, quoting [User Locations - Ultimate Member](https://ultimatemember.com/extensions/user-locations/),
> The Google Maps Platform is a paid service that comes with $200 in free usage each month.

But I don't know how much traffic we'll have. But I think we already bought Gsuite which provides the API key to Google map and use it to display location in a user profile.

>I am not sure how to integrate the layering information, i.e. Wat, Temples, Mahayana, Zen, etc, into UM's user location. Currently there is an option to display user roles marker, i.e. subscribers, admin, etc, and we may have to change them into something like "practitioners", "monk", "wat", "zen temples", etc.

According to [the documentation of User Locations Setup - Ultimate Member](https://docs.ultimatemember.com/article/1545-user-locations-setup#role_icons), we can also show avatar as the icon besides using role marker and we can also "choose fields(s) to display in marker's window". Can these options help resolve your concern?

Btw, personally I feel the User Locations extension is promising and your finding that the current page is created by GoogleMyMaps (I imagine this is a cumbersome process and we need to repeatedly go through the same process to update the map) gives us another big reason to switch to this extension. I haven't asked for Dhammarato and other stakeholderss input. Maybe after testing it locally, you can provide list pros and cons of different approaches and then set it up on OSF for other people to have an evaluation?

> 
> Let me know if it would be good to develop UM's user location, I think I can help on that. For the current issue, I think I would need an access to the GoogleMyMaps, or its associated csv files. I can perform a mapping from "user-update/?r=2125" to "user/santi-janati" and all other locations. Else, we just let the non-admin members wait a bit longer :P
> 
> Let me know what you think!

**Comment by anuutama** [comment]
I have tested the UM user location extension in the local server. It is pretty nice, as the user can pin the map and give their specific location (not just a combination of addresses as in the osf_googlemap field).

However, they seem to have some limitation in ways the map can render. First, a user can find and pin their location. Second, there is a map that can be generated from the UM's member directory. The map below displays the two user location that I edited and key manually.
![Member Map Screenshot](https://github.com/user-attachments/assets/b535615b-20df-490e-a09e-6a07788ebe6e)

There are three ways to display the pin: avatar (profile picture, which many temples do not have), user role (currently 95%++ are just subscribers), and normal default pin. I can see how this would work. First, all the different temples would need to have different roles, i.e. Wat, Zen Temples, etc. Then, either we need to manually input the location, or derive the location from the current googlemymaps via some programmatic means. Note that we can (and maybe should) also do this for the current temple profiles as well, as it would make it easier to keep track and update googlemymaps, UM's userlocation map, or use it to make any map in any platform. 

I can work on this, but at this point I need to be able to import and export user data easily. I would also need the csv that was used to generate the map from googlemymaps.

These are some pro and cons of the different maps.
Pros - GoogleMyMaps:
1. Relatively lightweight and static. Almost no lag experienced.
2. Can display different layers (which correspond to different styles of temple)
Cons - GoogleMyMaps:
1. No searching by name or location, user have to zoom in manually 
Pros - UM's user location:
1. Can search by name or location (two different search bars)
2. Up to date with the user/temple database
Cons - GoogleMyMaps:
1. A bit more laggy compared to GoogleMyMaps. This is because they will perform search every time the zoom level is changed.
2. Cannot display different layers. Need to identify different style of temples by the logo of the user role. We have to design different logos for Theravada's wats, zen temples, and so on.
